### PR TITLE
[chore] Update otel

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Updates the core modules to use the latest changes that remove component status feature from `component`.

All references to status features from `component` have been updated to `componentstatus`.

Unlike core's implementation of `SharedComponent`, contrib's implementation had no component status features.  I believe contrib's `ShareComponent` implementation should be updated, but that can now be added in a separate PR since it already wasn't supporting component status reporting.

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
@mwear I got healthcheckv2extension building and tests passing, but did not dig deeper than that.

**Review by dmitryax — APPROVED** [review]

**Review by mwear — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
We can merge this to unblock core tests and block the release via https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34723

**Pull request merged**
